api/core/v2/health.go

```diff
@@ -1,21 +1,40 @@
 package v2
 
 import (
+	"encoding/json"
+	"fmt"
+
 	"github.com/coreos/etcd/etcdserver/etcdserverpb"
 )
 
 // ClusterHealth holds cluster member status info.
 type ClusterHealth struct {
 	// MemberID is the etcd cluster member's ID.
 	MemberID uint64
+
+	// MemberIDHex is the hexadecimal representation of the member's ID.
+	MemberIDHex string
+
 	// Name is the cluster member's name.
 	Name string
-	// Err holds the string representation of any errors encountered while checking the member's health.
+
+	// Err contains any error encountered while checking the member's health.
 	Err string
+
 	// Healthy describes the health of the cluster member.
 	Healthy bool
 }
 
+func (h ClusterHealth) MarshalJSON() ([]byte, error) {
+	if h.MemberIDHex == "" {
+		h.MemberIDHex = fmt.Sprintf("%x", h.MemberID)
+	}
+	type Clone ClusterHealth
+	var clone *Clone
+	clone = (*Clone)(&h)
+	return json.Marshal(clone)
```

Comment:
Why is that needed? Wouldn't `return json.Marshal(h)` result in the same `[]byte`? Is that because it creates an infinite recursion of `h.ClusterHealth()`?

Reply:
Yes, the clone is required to avoid the infinite recursion.